sympy/stats/frv.py

```diff
@@ -191,34 +194,25 @@ def check(*args):
     @property
     @cacheit
     def dict(self):
+        if self.is_symbolic:
+            return Density(self)
         return dict((k, self.pdf(k)) for k in self.set)
 
-    @property
-    def pdf(self):
-        x = Symbol('x')
-        return Lambda(x, Piecewise(*(
-            [(v, Eq(k, x)) for k, v in self.dict.items()] + [(0, True)])))
-
-    @property
-    def characteristic_function(self):
-        t = Dummy('t', real=True)
-        return Lambda(t, sum(exp(I*k*t)*v for k, v in self.dict.items()))
-
-    @property
-    def moment_generating_function(self):
-        t = Dummy('t', real=True)
-        return Lambda(t, sum(exp(k * t) * v for k, v in self.dict.items()))
+    def pdf(self, *args): # to be overrided by specific distribution
+        pass
```

Comment:
if it's an abstract method, it should either raise an exception or use a metaclass/property for abstract methods.